product/tests/test_product.py

```diff
@@ -16,20 +17,33 @@ def product0():
     return Product(qr_code=QR, product_name=NAME, description=DESCRIPTION)
 
 
+@pytest.fixture
+def product1():
+    return Product(qr_code=QR1, product_name=NAME, description=DESCRIPTION)
+
+
 @pytest.fixture
 def saved_product0(product0):
     product0.save_product()
     return product0
 
 
+@pytest.fixture
+def saved_product1(product1):
+    product1.save_product()
+    return product1
+
+
 class TestProductModel:
     @pytest.mark.django_db()
-    def test_save_product_and_delete_product(self, saved_product0):
+    def test_save_product_and_delete_product(self, saved_product0, saved_product1):
         assert saved_product0 in Product.objects.all()
-        Product.delete_product(saved_product0.qr_code)
+        saved_product0.delete()
         assert saved_product0 not in Product.objects.all()
-        saved_product0.save_product()
-        assert saved_product0 in Product.objects.all()
+        with pytest.raises(Product.DoesNotExist):
+            test_doesnotexist = Product("DOESNOTEXIST", "TEST", "TEST")
+            test_doesnotexist.delete_product()
```

Comment:
This should be on a test function of its own, as it's not related to the delete-save cycle that we're testing, and while you do that, make the context manager include just the line that should raise the exception.

```
        test_doesnotexist = Product("DOESNOTEXIST", "TEST", "TEST")
        with pytest.raises(Product.DoesNotExist):
            test_doesnotexist.delete_product()
```

Reply:
fixed